src/main/java/org/commonwl/view/Scheduler.java

```diff
@@ -55,4 +62,40 @@ public void removeOldQueuedWorkflowEntries() {
         logger.info(queuedWorkflowRepository.deleteByTempRepresentation_RetrievedOnLessThanEqual(removeTime)
                 + " Old queued workflows removed");
     }
+
+
+    @Scheduled(cron = "${cron.clearTmpDir}")
+    public void clearTmpDir() {
+        // TODO: find source of tmp dir creation and delete from there
+
+        // wipe tmp dir and log info
+        try {
+            // Run command
+            String[] command = {"find", "/tmp", "-ctime", "+" + TMP_DIR_AGE_LIMIT_DAYS, "-exec", "rm", "-rf", "{}", "+"};
+            ProcessBuilder clearProcess = new ProcessBuilder(command);
+            logger.info("Clearing /tmp directory for content older than " + TMP_DIR_AGE_LIMIT_DAYS + " day" + (TMP_DIR_AGE_LIMIT_DAYS > 1 ? "s" : "") + "...");
+            Process process = clearProcess.start();
+
+            // Read output from the process using threads
+            StreamGobbler inputGobbler = new StreamGobbler(process.getInputStream());
+            StreamGobbler errorGobbler = new StreamGobbler(process.getErrorStream());
+            errorGobbler.start();
+            inputGobbler.start();
+
+            // Wait for process to complete
+            int exitCode = process.waitFor();
+            if (exitCode == 0) {
+                inputGobbler.join();
+                logger.info(inputGobbler.getContent());
+                logger.info("Successfully Cleared /tmp directory");
```

Comment:
s/Cleared/cleared for consistency with other messages? :grimacing: 

Also, once the `TODO` above about locating the tmp dir is solved, we should probably include the location in the logs to help in case a sysadmin needs to check the directory permissions, modes, etc.

Reply:
> s/Cleared/cleared for consistency with other messages? 😬
> 
> Also, once the `TODO` above about locating the tmp dir is solved, we should probably include the location in the logs to help in case a sysadmin needs to check the directory permissions, modes, etc.

Hi @kinow ,
The idea was to delete the file within the process that actually creates these files.
But that would be a mistake since the file may still be in use.

Another idea was to delete the file the moment it is no longer in use, but an exact time would be difficult to determine

Hence a solution that works would be to delete all files above a certain age.

so I would be removing the `TODO` remark.
But before I do that, do you have any comments on a much better approach?

Reply:
Sounds good to me. The other comment about using a Java native solution is still valid I think.

So instead of forking a new process we could try first using Java to iterate/walk the temp dir, filter for files/dirs older than the threshold, and finally delete or mark them for deletion (I assume we will try to delete it and in case it fails, call `deleteOnExit` or just log it... maybe Commons IO has something that we can use... just food for thought 👍 )